### PR TITLE
getLight() should initialize all fields.

### LINKED ISSUE #4014: lightbulb sample crash with args "-m -a vulkan"
**Describe the bug**
When i run lightbulb sample with args "-m -a vulkan", application crashed with output log :

VULKAN ERROR: (VUID-vkQueuePresentKHR-pWaitSemaphores-03268) Validation Error: [ VUID-vkQueuePresentKHR-pWaitSemaphores-03268 ] Object 0: handle = 0x225c05e9078, type = VK_OBJECT_TYPE_QUEUE; Object 1: handle = 0x68d1dd0000000007, type = VK_OBJECT_TYPE_SEMAPHORE; | MessageID = 0x251f8f7a | vkQueuePresentKHR: Queue VkQueue 0x225c05e9078[] is waiting on pWaitSemaphores[0] (VkSemaphore 0x68d1dd0000000007[]) that has no way to be signaled. The Vulkan spec states: All elements of the pWaitSemaphores member of pPresentInfo must reference a semaphore signal operation that has been submitted for execution and any semaphore signal operations on which it depends (if any) must have also been submitted for execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkQueuePresentKHR-pWaitSemaphores-03268)

it`s ok with args "-a vulkan".

**To Reproduce**
 ./lightbulb.exe -m -a vulkan /d/filament-main/filament-main/assets/models/monkey/monkey.obj

**Desktop (please complete the following information):**
 - OS: [e.g. windows ]
 - GPU: [e.g. NVIDIA RTX 3090]
 - Backend: [Vulkan]

### PR DESCRIPTION
**Pull request by prideout**
Fixes #4014.

### CODE REVIEW

**Review by romainguy — COMMENTED** [review]
Good catch!

**Review by romainguy — APPROVED** [review]

**Comment by prideout** [comment]
These fields are consumed in `evaluatePunctualLights` so I think they need to be initialized to something.

**Review by bejado — APPROVED** [review]

File: shaders/src/light_punctual.fs [hunk 1]
```diff
@@ -142,6 +142,11 @@ Light getLight(const uint index) {
         light.contactShadows = bool((shadowBits >> 1u) & 0x1u);
         light.shadowIndex = (shadowBits >> 2u) & 0xFu;
         light.shadowLayer = (shadowBits >> 6u) & 0xFu;
+    } else {
+        light.castsShadows = false;
+        light.contactShadows = false;
+        light.shadowIndex = 0;
+        light.shadowLayer = 0;
```
Comment (pixelflinger) [inline shaders/src/light_punctual.fs:149]: I wouldn't do a if-else for this -- just initialize the variables before the if.

Comment (prideout) [reply shaders/src/light_punctual.fs]: Done.  Also changed `0` to `0u` to (hopefully) fix the build errors seen on on some platforms.

Comment (pixelflinger) [reply shaders/src/light_punctual.fs]: gotta love glsl!

**Pull request merged**